corehq/apps/users/models.py

```diff
@@ -1112,6 +1112,9 @@ def is_dimagi(self):
         return self.username.endswith('@dimagi.com')
 
     def is_locked_out(self):
+        return self.supports_lockout() and self.should_be_locked_out()
+
+    def should_be_locked_out(self):
         return self.login_attempts >= MAX_LOGIN_ATTEMPTS
```

Comment:
Am I right to read this that in this PR every instance of `user.is_locked_out() and user.supports_lockout()` in the codebase was replaced with `user.is_locked_out()` and every other instance of `user.is_locked_out()` was replaced with `user.should_be_locked_out`? That's how I'm reading these changes, but I can't tell just from the PR whether _every_ usage was refactored to accommodate the repurposing of the name

Reply:
In the future to make it clearer, I might break this PR into a few prefactor commits followed by the main change:

Prefactors:
- rename `is_locked_out` and all its usages to `should_be_locked_out` (search the codebase to make sure no usages of is_locked_out remain)
- replace all usages of `user.should_be_locked_out() and user.supports_lockout()` with new method `user.is_locked_out()`
- Add test for refactor
Main change:
- Create metrics for failed logins

That makes it clearer to the reviewer what the process is and lets them evaluate the prefactor (where what we're looking for is to convince ourselves that nothing changed and nothing slipped through the cracks) independently from the new functionality (where what we care about is that the new thing works). No need to redo anything, just writing down how I might approach it in case that's a helpful pattern for you too

Reply:
> Am I right to read this that in this PR every instance of `user.is_locked_out() and user.supports_lockout()` in the codebase was replaced with `user.is_locked_out()` and every other instance of `user.is_locked_out()` was replaced with `user.should_be_locked_out`?

Basically correct -- every previous instance of `user.is_locked_out()` was paired with `user.supports_lockout()`. No other instances needed to be replaced with `user.should_be_locked_out` because none existed.

Reply:
> In the future to make it clearer, I might break this PR into a few prefactor commits followed by the main change:

Thanks! I'll definitely keep this in mind for future PRs. I do want to push back slightly, though, in that tests belong in the same commit as the code being changed. This allows the reviewer to ensure the changes are covered by tests, while encouraging the developer to think of the tests as he codes.